### ISSUE TITLE
Add changelog

### ISSUE THREAD

**Issue by johnjohndoe** [spring-projects/spring-android#33]
Please consider providing a textual **changelog** so it is easier for users and developers to see what changes from version to version.

* [Example changelog of devise](https://github.com/plataformatec/devise/blob/v3.1.1/CHANGELOG.md)
* [Example changelog of cancan](https://github.com/ryanb/cancan/blob/1.6.10/CHANGELOG.rdoc)
* [Example changelog of angular.js](https://github.com/angular/angular.js/blob/v1.1.5/CHANGELOG.md)
* [GitHub offers release notes](https://github.com/jeffkreeftmeijer/fuubar/releases/new)
* [GitHub release notes example of better_errors](https://github.com/charliesome/better_errors/releases)
* [Umweltzone](https://github.com/johnjohndoe/Umweltzone/blob/master/CHANGELOG.md)

**Comment by kihaki** [comment]
Yes please, especially documenting breaking changes if there are any. We are currently updating a major commercial project to a new version and while I appreciate that this project is free to use and open source the lack of a changelog makes it a very hard decision to update since we can never be sure an update won't break things and testing has to be very intensive.
This discourages developers from using the spring framework.

Adding a changelog, at least for breaking changes, would hugely benefit our work and I would be really happy to see it added.
